### PR TITLE
Implementing post mutations

### PR DESCRIPTION
**Pull request by samueldurantes**

### CODE REVIEW

File: packages/server/src/modules/post/mutations/postCreateMutation.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export const postCreate = mutationWithClientMutationId({
+  name: 'PostCreate',
+  inputFields: {
+    title: { type: new GraphQLNonNull(GraphQLString) },
+    body: { type: new GraphQLNonNull(GraphQLString) }
+  },
+  mutateAndGetPayload: async ({ title, body }, ctx: GraphQLContext) => {
+    if (!ctx?.user) {
+      throw new Error('You are not logged in!')
+    }
```
Comment (noghartt) [inline packages/server/src/modules/post/mutations/postCreateMutation.ts:17]: You could add it as a middleware.

**Review by noghartt — APPROVED** [review]

File: packages/server/src/modules/post/mutations/postCreateMutation.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    const post = new PostModel({
+      title,
+      body,
+      author: ctx.user
+    })
+
+    await Promise.all([
+      post.save(),
+      ctx.user.updateOne({
```
Comment (daniloab) [inline packages/server/src/modules/post/mutations/postCreateMutation.ts]: update one is deprecated

Comment (samueldurantes) [reply packages/server/src/modules/post/mutations/postCreateMutation.ts:27]: There is some alternative?

File: packages/server/src/modules/post/mutations/postCreateMutation.ts (hunk 1 above)
Comment (ssdurante) [reply packages/server/src/modules/post/mutations/postCreateMutation.ts:17]: I will improve this in the next pull request, thank you!

File: packages/server/src/modules/post/mutations/postCreateMutation.ts (hunk 2 above)
Comment (ssdurante) [reply packages/server/src/modules/post/mutations/postCreateMutation.ts]: Done!

File: packages/server/src/modules/post/mutations/postCreateMutation.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ export const postCreate = mutationWithClientMutationId({
 
     await Promise.all([
       post.save(),
-      ctx.user.updateOne({
+      ctx.user.update({
```
Comment (daniloab) [inline packages/server/src/modules/post/mutations/postCreateMutation.ts:27]: prefer
```
await UserModel.findOneAndUpdate({
 _id: ctx.user,
}, {
  $addToSet: { posts: post._id }
})
```

**Pull request merged**